plugin-api/src/main/java/org/sonar/api/server/rule/internal/ImpactMapper.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Sonar Plugin API
+ * Copyright (C) 2009-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.server.rule.internal;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import org.sonar.api.issue.impact.Severity;
+import org.sonar.api.issue.impact.SoftwareQuality;
+import org.sonar.api.rules.RuleType;
+
+import static org.sonar.api.rule.Severity.BLOCKER;
+import static org.sonar.api.rule.Severity.CRITICAL;
+import static org.sonar.api.rule.Severity.INFO;
+import static org.sonar.api.rule.Severity.MAJOR;
+import static org.sonar.api.rule.Severity.MINOR;
+
+/**
+ * @deprecated since 10.1 This is only used for mapping deprecated types and severities until they are removed
+ */
+@Deprecated(since = "10.1")
```

Comment:
It's odd to add something that is already deprecated. Do plugins need to use this class? Could we just make it package-private?

Reply:
For Adhoc rule, we will need to do the mapping and backmapping also in product side (Since we only provide an interface for it), so we need it to be public.
The idea we wanted to convey was that it will disappear at the same time that the old type and severity are removed from the plugin-api

Reply:
I think this is the kind of class that we could put in a common library as we already discussed. It's common code for implementing products and not an interface between plugins and products. But it's probably out of scope for your sprint